cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/reachability/ReachableBlock.qll

```diff
@@ -35,6 +38,14 @@ class ReachableBlock extends IRBlock {
   ReachableBlock() {
     isBlockReachable(this)
   }
+
+  final ReachableBlock getAFeasiblePredecessor() {
+    result = getAFeasiblePredecessorBlock(this)
+  }
+
+  final ReachableBlock getAFeasibleSuccessor() {
+    this = getAFeasiblePredecessorBlock(result)
+  }
```

Comment:
This is a dangerous API. It's easy to accidentally call `getASuccessor` instead of `getAFeasibleSuccessor`. How much of a hassle would it be to let `ReachableBlock` extend `TIRBlock` instead of `IRBlock` so it doesn't inherit `getASuccessor`? If it's not convenient to expose `TIRBlock`, an alternative is to make an `IRBlockBase` class like I've done with `ControlFlowNodeBase`.

Reply:
Good suggestion. I've factored out `IRBlockBase`, and used it as the base type for both `IRBlock` and `ReachableBlock`.